youtokentome/cpp/yttm.pyx

```diff
@@ -130,12 +132,30 @@ cdef class BPE:
             raise ValueError(status.message.decode())
         return subword.decode()
 
-    def decode(self, ids):
-        assert isinstance(ids, list)
+    def decode(self, ids, ignore_ids):
+
+        if not isinstance(ids, list):
+            raise TypeError(
+                "ids have to be a list instance but {} found".format(type(list))
+            )
+
+        if not (
+            isinstance(ignore_ids, Collection) or ignore_ids is None
```

Comment:
`if not isinstance(ignore_ids, Collection) and ignore_ids is not None` -- much easy to read.

Reply:
Fixed